test/test_buffer_protocol.py

```diff
@@ -0,0 +1,163 @@
+            offset = first * get_dtype_size(dtype)
+        last = first + count if count > 0 else len(arr)
+
+        pyarray = array.array(TORCH_TO_ARRAYTYPE[dtype], arr)
+        tensor = torch.from_buffer(pyarray, dtype=dtype, count=count, offset=offset, **kwargs)
+
+        self.assertSequenceEqual(pyarray[first:last], tensor)
+        return (pyarray, tensor)
+
+    @onlyCPU
+    @dtypes(*all_types())
+    def test_same_type(self, device, dtype):
+        self._run_test(dtype, [1])
+        self._run_test(dtype, [0, 1, 2, 3])
+        self._run_test(dtype, [0] * 100)
+
+    @onlyCPU
+    @dtypes(*all_types())
+    def test_requires_grad(self, device, dtype):
+        def _run_test_and_check_grad(requires_grad, *args, **kwargs):
+            kwargs["requires_grad"] = requires_grad
+            _, tensor = self._run_test(*args, **kwargs)
+            self.assertTrue(tensor.requires_grad == requires_grad)
+        if dtype.is_floating_point or dtype.is_complex:
+            _run_test_and_check_grad(True, dtype, [1])
+            _run_test_and_check_grad(True, dtype, [0, 1, 2, 3])
+            _run_test_and_check_grad(True, dtype, [0] * 100)
+        _run_test_and_check_grad(False, dtype, [1])
+        _run_test_and_check_grad(False, dtype, [0, 1, 2, 3])
+        _run_test_and_check_grad(False, dtype, [0] * 100)
+
+    @onlyCPU
+    @dtypes(*all_types())
+    def test_with_offset(self, device, dtype):
+        input = self.INPUT
+        # Offset should be valid whenever there is, at least,
+        # one remaining element
+        for i in range(len(input)):
+            self._run_test(dtype, input, first=i)
+
+    @onlyCPU
+    @dtypes(*all_types())
+    def test_with_count(self, device, dtype):
+        input = self.INPUT
+        # Count should be valid for any valid in the interval
+        # [-1, len(input)], except for 0
+        for i in range(-1, len(input) + 1):
+            if i != 0:
+                self._run_test(dtype, input, count=i)
+
+    @onlyCPU
+    @dtypes(*all_types())
+    def test_with_count_and_offset(self, device, dtype):
+        input = self.INPUT
+        # Explicit default count [-1, 1, 2, ..., len]
+        for i in range(-1, len(input) + 1):
+            if i != 0:
+                self._run_test(dtype, input, count=i)
+        # Explicit default offset [0, 1, ..., len - 1]
+        for i in range(len(input)):
+            self._run_test(dtype, input, first=i)
+        # All possible combinations of count and dtype aligned
+        # offset for 'input'
+        # count:[1, 2, ..., len - 1] x first:[0, 1, ..., len - count]
+        for i in range(1, len(input)):
+            for j in range(len(input) - i + 1):
+                self._run_test(dtype, input, count=i, first=j)
+
+    @onlyCPU
+    @dtypes(*all_types())
+    def test_invalid_positional_args(self, device, dtype):
+        input = self.INPUT
+        bytes = get_dtype_size(dtype)
+        # Empty array
+        with self.assertRaises(ValueError):
```

Comment:
Could you use `assertRaisesRegex` and provide a distinctive part of the error message for these? Just to make sure that you're not catching an error that is not the one you expect.